test_migration_restart test fail

https://github.com/dragonflydb/dragonfly/actions/runs/24622859828

```
        nodes[0].migrations = []
        nodes[0].slots = [(0, final_migration_range[0] - 1), (final_migration_range[1] + 1, 16383)]
        nodes[1].slots = [final_migration_range]
        await push_config(json.dumps(generate_config(nodes)), [node.admin_client for node in nodes])
    
>       assert await seeder.compare(capture, nodes[0].instance.port)
E       assert False

```

--- a/src/server/cluster/cluster_family.cc
+++ b/src/server/cluster/cluster_family.cc
@@ -593,7 +593,8 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, CommandContext* cmd_cntx)
 
     new_config = new_config->CloneWithChanges(enable_slots, disable_slots);
 
-    StartNewSlotMigrations(*new_config);
+    // Capture prev config before SetCurrent so StartNewSlotMigrations can diff correctly.
+    auto prev_config = ClusterConfig::Current();
 
     SlotSet before =
         ClusterConfig::Current() ? ClusterConfig::Current()->GetOwnedSlots() : SlotSet(true);
@@ -632,6 +633,10 @@ void ClusterFamily::DflyClusterConfig(CmdArgList args, CommandContext* cmd_cntx)
           << "Flushing newly unowned slots: " << deleted_slots.ToString();
       WriteFlushSlotsToJournal(deleted_slots);
     }
+
+    // Start new migrations only after stale data is flushed. This ensures DFLYMIGRATE FLOW
+    // cannot write data that is immediately wiped by DeleteSlots above.
+    StartNewSlotMigrations(*new_config, prev_config);
   }
 
   return cmd_cntx->SendOk();
@@ -744,10 +749,11 @@ void ClusterFamily::DflyClusterFlushSlots(CmdArgList args, CommandContext* cmd_c
   return cmd_cntx->SendOk();
 }
 
-void ClusterFamily::StartNewSlotMigrations(const ClusterConfig& new_config) {
+void ClusterFamily::StartNewSlotMigrations(const ClusterConfig& new_config,
+                                           const std::shared_ptr<ClusterConfig>& prev_config) {
   // TODO Add validating and error processing
-  auto out_migrations = new_config.GetNewOutgoingMigrations(ClusterConfig::Current());
-  auto in_migrations = new_config.GetNewIncomingMigrations(ClusterConfig::Current());
+  auto out_migrations = new_config.GetNewOutgoingMigrations(prev_config);
+  auto in_migrations = new_config.GetNewIncomingMigrations(prev_config);
 
   util::fb2::LockGuard lk(migration_mu_);
 

--- a/src/server/cluster/cluster_family.h
+++ b/src/server/cluster/cluster_family.h
@@ -99,7 +99,8 @@ class ClusterFamily {
   std::shared_ptr<IncomingSlotMigration> GetIncomingMigration(std::string_view source_id)
       ABSL_LOCKS_EXCLUDED(migration_mu_);
 
-  void StartNewSlotMigrations(const ClusterConfig& new_config);
+  void StartNewSlotMigrations(const ClusterConfig& new_config,
+                              const std::shared_ptr<ClusterConfig>& prev_config);
 
   // must be destroyed excluded set_config_mu and migration_mu_ locks
   struct PreparedToRemoveOutgoingMigrations {

